db/trie/branchroottrie.go

```diff
@@ -53,22 +61,65 @@ func (tr *branchRootTrie) Stop(_ context.Context) error {
 	return nil
 }
 
+func (tr *branchRootTrie) RootNode() Node {
+	return tr.root
+}
+
+func (tr *branchRootTrie) SetRootNode(root Node) error {
+	bn, ok := root.(*branchNode)
+	if !ok {
+		return errors.Wrapf(ErrInvalidTrie, "root should be a branch")
+	}
+	tr.resetRoot(bn)
+
+	return nil
+}
+
 func (tr *branchRootTrie) RootHash() []byte {
-	return tr.rootHash
+	h, err := tr.root.hash(tr, false)
+	if err != nil {
+		panic("failed to get hash of trie")
+	}
+	return h
+}
+
+func (tr *branchRootTrie) Commit() ([]byte, error) {
+	h, err := tr.root.hash(tr, true)
+	if err == nil {
+		tr.rootHash = make([]byte, len(h))
+		copy(tr.rootHash, h)
+	}
+	return h, err
+}
+
+func (tr *branchRootTrie) Clone() Trie {
+	rootHash := make([]byte, len(tr.rootHash))
+	copy(rootHash, tr.rootHash)
+
```

Comment:
i recall you said now it's saving internal nodes, not serialized bytes in batch. How are these internal nodes saved in Clone()?

Reply:
As synced offline, this problem has been resolved.